File: 3195-find-the-minimum-area-to-cover-all-ones-i/3195-find-the-minimum-area-to-cover-all-ones-i.cpp
class Solution {
public:
    int minimumArea(vector<vector<int>>& grid) {
        int m = grid.size();
        int n = grid[0].size();
        int minimum_x = n - 1;
        int maximum_x = 0;
        int minimum_y = m - 1;
        int maximum_y = 0;
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (grid[i][j] == 1) {
                    minimum_x = min(minimum_x, j);
                    maximum_x = max(maximum_x, j);
                    minimum_y = min(minimum_y, i);
                    maximum_y = max(maximum_y, i);
                }
            }
        }
        return (maximum_x - minimum_x + 1) * (maximum_y - minimum_y + 1);
    }
};

File: 3195-find-the-minimum-area-to-cover-all-ones-i/3195-find-the-minimum-area-to-cover-all-ones-i.py
class Solution:
    def minimumArea(self, grid: List[List[int]]) -> int:
        rowMax = colMax = float('-inf')
        rowMin = colMin = float('inf')
        n = len(grid)
        m = len(grid[0])
        for i in range(n):
            for j in range(m):
                if grid[i][j]==1:
                    rowMin = min(rowMin,i)
                    rowMax = max(rowMax,i)
                    colMin = min(colMin,j)
                    colMax = max(colMax,j)
        return (rowMax - rowMin + 1)*(colMax - colMin + 1) if rowMax!=float('-inf') else 0
        